foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiToolChoice.java

```diff
@@ -0,0 +1,47 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionNamedToolChoice;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionNamedToolChoiceFunction;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionToolChoiceOption;
+import java.util.function.Supplier;
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * OpenAi ToolChoice to specify whether to call which tool.
+ *
+ * @since 1.4.0
+ */
+@RequiredArgsConstructor(access = AccessLevel.PROTECTED)
+public class OpenAiToolChoice {
+  @Nonnull final Supplier<? extends ChatCompletionToolChoiceOption> toolChoice;
```

Comment:
```suggestion
  @Nonnull final ChatCompletionToolChoiceOption toolChoice;
```
Could this directly be defined instead of supplier?